bitswap/network/httpnet/httpnet.go

```diff
@@ -0,0 +1,378 @@
+	log.Debugf("SendMessage: %s. %s", p, msg)
+	// todo opts
+	sender, err := bsnet.NewMessageSender(ctx, p, nil)
+	if err != nil {
+		return err
+	}
+	defer sender.Close()
+	return sender.SendMsg(ctx, msg)
+}
+
+func (bsnet *httpnet) Self() peer.ID {
+	return bsnet.host.ID()
+}
+
+func (bsnet *httpnet) Connect(ctx context.Context, p peer.AddrInfo) error {
+	log.Debugf("Connect: %s", p)
+	htaddrs, _ := network.SplitHTTPAddrs(p)
+	if len(htaddrs.Addrs) == 0 {
+		return nil
+	}
+	bsnet.host.Peerstore().AddAddrs(p.ID, htaddrs.Addrs, peerstore.PermanentAddrTTL)
+	for _, r := range bsnet.receivers {
+		r.PeerConnected(p.ID)
+	}
+	return nil
+}
+
+func (bsnet *httpnet) DisconnectFrom(ctx context.Context, p peer.ID) error {
+	return nil
+}
+
+func (bsnet *httpnet) Stats() network.Stats {
+	return network.Stats{
+		MessagesRecvd: atomic.LoadUint64(&bsnet.stats.MessagesRecvd),
+		MessagesSent:  atomic.LoadUint64(&bsnet.stats.MessagesSent),
+	}
+}
+
+func (nbsnet *httpnet) TagPeer(p peer.ID, tag string, w int) {
+}
+func (bsnet *httpnet) UntagPeer(p peer.ID, tag string) {
+}
+
+func (bsnet *httpnet) Protect(p peer.ID, tag string) {
+}
+func (bsnet *httpnet) Unprotect(p peer.ID, tag string) bool {
+	return false
+}
+
+func (bsnet *httpnet) NewMessageSender(ctx context.Context, p peer.ID, opts *network.MessageSenderOpts) (network.MessageSender, error) {
+	log.Debugf("NewMessageSender: %s", p)
+	pi := bsnet.host.Peerstore().PeerInfo(p)
+	urls := network.ExtractURLsFromPeer(pi)
+	if len(urls) == 0 {
+		return nil, ErrNoHTTPAddresses
+	}
+
+	return &httpMsgSender{
+		// ctx ??
+		peer:      p,
+		urls:      urls,
+		client:    bsnet.client,
+		receivers: bsnet.receivers,
+		closing:   make(chan struct{}, 1),
+		// opts: todo
+	}, nil
+}
+
+type httpMsgSender struct {
+	client    *http.Client
+	peer      peer.ID
+	urls      []*url.URL
+	receivers []network.Receiver
+	opts      network.MessageSenderOpts
+	closing   chan struct{}
+	closeOnce sync.Once
+}
+
+func (sender *httpMsgSender) SendMsg(ctx context.Context, msg bsmsg.BitSwapMessage) error {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	bsresp := msg.Clone()
+
+	go func() {
+		for range sender.closing {
+			cancel()
+		}
+	}()
+
+	sendErrors := func(err error) {
+		for _, recv := range sender.receivers {
+			recv.ReceiveError(err)
+		}
+	}
+
+	sendURL, _ := url.Parse(sender.urls[0].String())
+	sendURL.RawQuery = "format=raw&dag-scope=block"
+
+	// TODO: assuming we don't have to manage making concurrent
+	// requests here.
+	for _, entry := range msg.Wantlist() {
+		var method string
+		switch {
+		case entry.Cancel:
+			continue // todo: handle cancelling ongoing
+		case entry.WantType == pb.Message_Wantlist_Block:
+			method = "GET"
+		case entry.WantType == pb.Message_Wantlist_Have:
+			method = "HEAD"
+		default:
+			continue
+		}
+
+		sendURL.Path = "/ipfs/" + entry.Cid.String()
+		headers := make(http.Header)
```

Comment:
Proactively set User-Agent to something like  https://github.com/ipfs/boxo/blob/058422d394b286dc18a24ba6e3f0c45f05e4c6c0/routing/http/client/transport.go#L60 so we see when people  start using this in their own code? 

(go-libp2p does similar thing so when people run with defaults we see their repo name in libp2p identify)

```suggestion
		headers := make(http.Header)
		headers.Set("User-Agent", moduleVersion())
```

This way server owner will know who to blame for any client related issues.